packages/components/Menu/src/MenuList/MenuList.tsx

```diff
@@ -58,17 +59,27 @@ export const MenuList = compose<MenuListType>({
     }, []);
 
     return (_final: MenuListProps, children: React.ReactNode) => {
-      const filteredChildren = React.Children.toArray(children).filter((child) => React.isValidElement(child));
-      const childrenWithSet = filteredChildren.map((child, index: number) => {
-        return React.cloneElement(
-          child as React.ReactElement<unknown, string | React.JSXElementConstructor<any>>,
-          {
-            accessibilityPositionInSet: index + 1, // win32
-            accessibilityPosInSet: index, // windows
-            accessibilitySetSize: filteredChildren.length, //win32
-            accessibilitySizeOfSet: filteredChildren.length, //windows
-          } as any,
-        );
+      const itemCount = React.Children.toArray(children).filter(
```

Comment:
Maybe store the result of the filtered array so you can reuse it to create childrenWithSet?

Reply:
The reason I didn't do this is because we're filtering out MenuDividers. If I were to reuse this array, any MenuDivider wouldn't be included in childrenWithSet.

Reply:
Ah, that makes sense.